pyccel/codegen/printing/ccode.py

```diff
@@ -368,7 +368,7 @@ def _print_IfTernaryOperator(self, expr):
         cond = self._print(expr.cond)
         value_true = self._print(expr.value_true)
         value_false = self._print(expr.value_false)
-        return '({cond}) ? {true} : {false}'.format(cond = cond, true =value_true, false = value_false)
+        return '(({cond}) ? {true} : {false})'.format(cond = cond, true =value_true, false = value_false)
```

Comment:
This problem is not specific to C. The issue mentions changing the inheritance of `IfTernaryOperator` but I do not see that here